SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -306,6 +306,13 @@ void USpatialNetDriver::OnConnectionToSpatialOSSucceeded()
 	Connection = ConnectionManager->GetWorkerConnection();
 	check(Connection);
 
+	// If the current Connection comes from an outdated ClientTravel, the associated NetDriver (this) won't match
+	// the NetDriver from the Engine, resulting in a crash. Here, if the NetDriver is outdated, we leave the callback.
+	if (GEngine->GetWorldContextFromPendingNetGameNetDriver(this) == nullptr)
+	{
```

Comment:
worth adding some verbose logging in here?

Reply:
I still have one last thing to check (unbind the callback), but indeed that could be a good thing to log, thanks 👍

Reply:
I suggest verbose since the related slack thread Stavka says they are running this every tick.

Reply:
Wouldn't warning be a better option? This is skipping a whole connection setup because a NetDriver is outdated, which probably highlights unwanted/unnecessary ClientTravels (like what happened to Stavka)

Reply:
Warnings will be annoying if that is part of how they want to engineer it, not sure if we want to discourage it, lacking some context on the issue. @mironec might have a view since he was involved in the discussion?

Reply:
Well, that was actually a bug that Stavka encountered. The thing is that here, every ClientTravel going through this will be entirely skipped, so I'm not sure it could really be an intended path. Let's see with Miron, Josh or Jess's pov on this

Reply:
let's chat with the people that raised this bug and see what they say

Reply:
I would say let's make this a warning to begin with and see how our customers take it. This bug specifically affects a customer and it would be interesting to see what they do with the warning and if they implement game-side fixes to avoid the warning. Or they request us to deal with it.

Reply:
Sahil seems to prefer a Warning as the 1 Client Travel per tick was not intended